finetrainers/trainer.py

```diff
@@ -364,34 +578,46 @@ def train(self) -> None:
                 logs = {}
 
                 with accelerator.accumulate(models_to_accumulate):
-                    videos = batch["videos"]
-                    prompts = batch["prompts"]
-                    batch_size = len(prompts)
-
-                    if self.args.caption_dropout_technique == "empty":
-                        if random.random() < self.args.caption_dropout_p:
-                            prompts = [""] * batch_size
+                    if not self.args.precompute_conditions:
+                        videos = batch["videos"]
+                        prompts = batch["prompts"]
+                        batch_size = len(prompts)
+
+                        if self.args.caption_dropout_technique == "empty":
+                            if random.random() < self.args.caption_dropout_p:
+                                prompts = [""] * batch_size
+
+                        latent_conditions = self.model_config["prepare_latents"](
+                            vae=self.vae,
+                            image_or_video=videos,
+                            patch_size=self.transformer_config.patch_size,
+                            patch_size_t=self.transformer_config.patch_size_t,
+                            device=accelerator.device,
+                            dtype=weight_dtype,
+                            generator=generator,
+                        )
+                        other_conditions = self.model_config["prepare_conditions"](
+                            tokenizer=self.tokenizer,
+                            text_encoder=self.text_encoder,
+                            tokenizer_2=self.tokenizer_2,
+                            text_encoder_2=self.text_encoder_2,
+                            prompt=prompts,
+                            device=accelerator.device,
+                            dtype=weight_dtype,
+                        )
+                    else:
+                        latent_conditions = batch["latent_conditions"]
+                        other_conditions = batch["other_conditions"]
```

Comment:
`other_conditions` seem a bit unintuitive in this context. Prefer explicit naming.

Reply:
What would you prefer? `text_conditions` works?

Reply:
`text_conditions` in case when we're dealing them and `conditional_image_latents` when it's image-to-video.

Reply:
I will go with `text_conditions` and `latent_conditions` for the moment. We don't support image-to-video at the moment, but I would maybe combine image/control all under a single name because they will effectively just be latents of some kind